### PR TITLE
Don't crash kvs that can't be parsed.

### PR DESCRIPTION
**Pull request by ConnorRigby**
This only prevents crashing. It doesn't actually
properly parse the message causing the crash in #15

### CODE REVIEW

File: lib/nerves_wpa_supplicant/messages.ex [hunk 1]
```diff
@@ -150,8 +150,14 @@ defmodule Nerves.WpaSupplicant.Messages do
...
     |> String.split("\n", trim: true)
     |> List.foldl(%{}, fn pair, acc ->
-      [key, value] = String.split(pair, "=")
-      Map.put(acc, String.to_atom(key), kv_value(String.trim_trailing(value)))
+      case String.split(pair, "=") do
+        [key, value] ->
+          acc
+          |> Map.put(String.to_atom(key), kv_value(String.trim_trailing(value)))
+
+        _ ->
```
Comment (fhunleth) [inline lib/nerves_wpa_supplicant/messages.ex:160]: Would it be useful to log the offending string?

Comment (ConnorRigby) [reply lib/nerves_wpa_supplicant/messages.ex:160]: I think it would be. I'll add this

Comment (aaronrenner) [reply lib/nerves_wpa_supplicant/messages.ex:160]: I'm not sure which network it's balking at. Here are a couple:

```
DIRECT-0F-HP OfficeJetPro 8710
better-than-going-to-the-library
```

I can test it out tonight. Is there a way to tell from the log message which one is crashing it?

Comment (ConnorRigby) [reply lib/nerves_wpa_supplicant/messages.ex:160]: @aaronrenner I added a log message. I'd appreciate if you can try this out. 
It's likely some P2P device, such as a Roku, printer, TV etc

Comment (aaronrenner) [reply lib/nerves_wpa_supplicant/messages.ex:160]: Cool, thanks! I'll try it out this evening or first thing tomorrow morning.

File: lib/nerves_wpa_supplicant/messages.ex [hunk 2]
```diff
@@ -150,8 +150,19 @@ defmodule Nerves.WpaSupplicant.Messages do
...
+          Logger.debug([
+            "Failed to decode response: ",
+            inspect(resp, limit: :infinity)
```
Comment (fhunleth) [inline lib/nerves_wpa_supplicant/messages.ex:163]: lol. `limit: :infinity`. the pain!

**Review by fhunleth — APPROVED** [review]

File: lib/nerves_wpa_supplicant/messages.ex (hunk 2 above)
Comment (fhunleth) [reply lib/nerves_wpa_supplicant/messages.ex:163]: p.s. require Logger.

Comment (ConnorRigby) [reply lib/nerves_wpa_supplicant/messages.ex:163]: :man_facepalming:

**Pull request merged**